realm/src/main/java/io/realm/internal/Group.java

```diff
@@ -287,17 +253,27 @@ public String toJson() {
         return nativeToJson(nativePtr);
     }
 
-    protected native String nativeToJson(long nativeGroupPtr);
-
     public String toString() {
         return nativeToString(nativePtr);
     }
 
-    protected native void nativeCommit(long nativeGroupPtr);
-
-    protected native String nativeToString(long nativeGroupPtr);
-
     private void throwImmutable() {
         throw new IllegalStateException("Objects cannot be changed outside a transaction; see beginTransaction() for details.");
     }
+
+    protected native long createNative();
+    protected native long createNative(String filepath, int value);
+    protected native long createNative(byte[] data);
+    protected native long createNative(ByteBuffer buffer);
+    protected static native void nativeClose(long nativeGroupPtr);
```

Comment:
A `native` is not always static. According to the JNI specification, the second argument (the `jobject`) in the C/native implementation of the method depends on wether it is a static method or not. Source: http://docs.oracle.com/javase/1.5.0/docs/guide/jni/spec/design.html#wp715